datacats/cli/manage.py

```diff
@@ -70,28 +72,41 @@ def reload_(environment, opts):
     """Reload environment source and configuration
 
 Usage:
-  datacats reload [-b] [-p|--no-watch] [--syslog] [-s NAME] [--address=IP] [ENVIRONMENT [PORT]]
-  datacats reload -r [-b] [--syslog] [-s NAME] [--address=IP] [ENVIRONMENT]
+  datacats reload [-b] [-p|--no-watch] [--syslog] [-s NAME] [--site-url=SITE_URL]
+                            [--address=IP] [ENVIRONMENT [PORT]]
+  datacats reload -r [-b] [--syslog] [-s NAME] [--address=IP] [--site-url=SITE_URL]
+                            [ENVIRONMENT]
 
 Options:
-  --address=IP       Address to listen on (Linux-only) [default: 127.0.0.1]
-  -b --background    Don't wait for response from web server
-  --no-watch         Do not automatically reload templates and .py files on change
-  -p --production    Reload with apache and debug=false
-  -r --remote        Reload DataCats.com cloud instance
-  -s --site=NAME     Specify a site to reload [default: primary]
-  --syslog           Log to the syslog
+  --address=IP          Address to listen on (Linux-only) [default: 127.0.0.1]
+  --site-url=SITE_URL   The site_url to use in API responses. Can use Python template syntax
+                        to insert the port and address (e.g. http://{address}:{port}/)
+  -b --background       Don't wait for response from web server
+  --no-watch            Do not automatically reload templates and .py files on change
+  -p --production       Reload with apache and debug=false
+  -r --remote           Reload DataCats.com cloud instance
+  -s --site=NAME        Specify a site to reload [default: primary]
+  --syslog              Log to the syslog
 
 ENVIRONMENT may be an environment name or a path to an environment directory.
 Default: '.'
 """
     environment.require_data()
     environment.stop_ckan()
-    if opts['PORT'] or opts['--address'] != '127.0.0.1':
+    if opts['PORT'] or opts['--address'] != '127.0.0.1' or opts['--site-url']:
         if opts['PORT']:
             environment.port = int(opts['PORT'])
         if opts['--address'] != '127.0.0.1':
             environment.address = opts['--address']
+        if opts['--site-url']:
+            site_url = opts['--site-url']
+            # TODO: Check it against a regex or use urlparse
+            try:
+                site_url = site_url.format(address=environment.address, port=environment.port)
```

Comment:
Nopers, this is just so that users who need to know the port can use it in specifying their site url like this:

datacats create mysite --site-url "http://my-random-site.com:{port}/"

(Datacats will substitute the correct port in for "port" here